gitlab/client.py

```diff
@@ -1199,12 +1201,26 @@ def _query(
         self._total: Optional[str] = result.headers.get("X-Total")
 
         try:
-            self._data: List[Dict[str, Any]] = result.json()
+            data: List[Dict[str, Any]] = result.json()
         except Exception as e:
             raise gitlab.exceptions.GitlabParsingError(
                 error_message="Failed to parse the server message"
             ) from e
 
+        self._data = []
+        for item in data:
```

Comment:
I think this should not be the default behavior, but conditional on an argument we can supply. For example something like `if remove_duplicates:`.

Reply:
If this were a feature offered an an option I have a hard time thinking anyone would opt to have the duplicates returned.